framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -339,106 +339,125 @@ public TransferFunction createFlowTransferFunction(CFAbstractAnalysis<Value, Sto
+     * is not used, the defaults for checked code will be applied to locations without a default for unchecked code.
+     * <p>
+     * Subclasses may override this method to add defaults that cannot be specified with
+     * a {@code @DefaultInUncheckedCodeFor} or {@code @DefaultQualifierInHierarchyInUncheckedCode} meta-annotations or to change
+     * the standard defaults.
+     *
+     * @param defs QualifierDefault object to which defaults are added
+     */
+    protected void addUncheckedCodeDefaults(QualifierDefaults defs) {
+        for (Class<? extends Annotation> annotation : getSupportedTypeQualifiers()) {
+            DefaultInUncheckedCodeFor defaultInUncheckedCodeFor = annotation.getAnnotation(DefaultInUncheckedCodeFor.class);
+
+            if (defaultInUncheckedCodeFor != null) {
+                final DefaultLocation[] locations = defaultInUncheckedCodeFor.value();
+                defs.addUncheckedCodeDefaults(AnnotationUtils
+                                                      .fromClass(elements,
+                                                                        annotation),
+                                                     locations);
             }
-            Set<? extends AnnotationMirror> bottoms = this.qualHierarchy.getBottomAnnotations();
-            for (AnnotationMirror bot : bottoms) {
-                defs.addUncheckedCodeDefault(bot, DefaultLocation.PARAMETERS);
-                defs.addUncheckedCodeDefault(bot, DefaultLocation.LOWER_BOUNDS);
-                defs.addUncheckedCodeDefault(bot, DefaultLocation.FIELD);
-                // TODO: this isn't simply DefaultLocation.OTHERWISE, because that would
-                // also apply to type declarations, which isn't currently working as desired.
-                // See: https://groups.google.com/d/msg/checker-framework-dev/vk2V6ZFKPLk/v3hENw-e7gsJ
+
+            if (annotation.getAnnotation(DefaultQualifierInHierarchyInUncheckedCode.class) != null) {
+                if (defaultInUncheckedCodeFor != null) {
+                    // A type qualifier should either have a DefaultInUncheckedCodeFor or
+                    // a DefaultQualifierInHierarchyInUncheckedCode annotation.
+                    ErrorReporter
+                            .errorAbort("GenericAnnotatedTypeFactory.createQualifierDefaults: " +
+                                                "qualifier has both @DefaultInUncheckedCodeFor and @DefaultQualifierInHierarchyInUncheckedCode annotations: "
+                                                + annotation.getCanonicalName());
+                } else {
+                    defs.addUncheckedCodeDefault(AnnotationUtils.fromClass(elements, annotation),
+                                                      DefaultLocation.OTHERWISE);
+                }
             }
         }
+        Set<? extends AnnotationMirror> tops = this.qualHierarchy.getTopAnnotations();
```

Comment:
Similarly to the checked code, shouldn't there be a check for `foundOtherwise` and use Unqualified if no other default is given? Ah, is that not needed because of the `addUncheckedStandardDefaults`?
Add a comment that this is not needed in this case?